lss

```diff
@@ -37,6 +37,68 @@ import logging
 import pyseq
 
 
+def tree(source, level, seq_format):
```

Comment:
This isn't as pretty as it could be, but I was trying to do it without iterating over everything too many times, and without adding a bunch of new objects (Nodes).